src/lib/graphqlFetcher.ts

```diff
@@ -0,0 +1,23 @@
+export default async function graphqlFetcher<T>(
+  endpoint: string,
+  query: string,
+  variables?: object
+): Promise<T | undefined> {
+  try {
+    const response = await fetch(endpoint, {
+      body: JSON.stringify({ query, variables }),
+      headers: { "Content-type": "application/json" },
+      method: "POST",
+    });
+
+    if (!response.ok) {
+      throw new Error(`HTTP error: ${response.status}`);
+    }
+
+    const { data } = await response.json();
```

Comment:
It's a good idea to put it in a separate function, but I don't really understand why not to use gql client?

also, should this function throw the error and not just legitimize? otherwise we won't get an "error" in useSwr?